providers/gupshup/src/lib/gupshup.provider.ts

```diff
@@ -46,9 +46,9 @@ export class GupshupSmsProvider implements ISmsProvider {
     const opts = {
       method: 'POST',
       headers: {
-        'Content-Type': 'application/json',
+        'Content-Type': 'application/x-www-form-urlencoded',
       },
-      body: JSON.stringify(params),
+      body: new URLSearchParams(params),
```

Comment:
shouldn't we send the data in the URL according to their [documentation](https://docs.gupshup.io/docs/send-message-to-single-number#send-api-request)? and I'm not sure about the POST method.
Btw the URL should start with `https`.

Reply:
```js
var request = require("request");
var options = { method: 'POST',
url: 'https://enterprise.smsgupshup.com/GatewayAPI/rest',
form:
{ method: 'sendMessage',
send_to: '919820XXXXXX',
msg: 'This is sample test message from GupShup',
msg_type: 'TEXT',
userid: 'XXXXXX',auth_scheme: 'PLAIN',
password: 'XXXXX',
format: 'JSON' } };
request(options, function (error, response, body) {
if (error) throw new Error(error);
console.log(body);
});
```

```python

import requests
url = "https://enterprise.smsgupshup.com/GatewayAPI/rest"
payload =
"method=sendMessage&send_to=919820XXXXXX&msg=This%20is%20sample%20test%20message%20from%20
GupShup&msg_type=TEXT&userid=XXXXXX&auth_scheme=PLAIN&password=XXXXX&format=JSON"
response = requests.request("POST", url, data=payload)
print(response.text)

```

@LetItRock They support both `GET` and `POST` method. I took the example from their `Node` and `Python` codes.
Should we update it to use `GET` instead?

Reply:
@BiswaViraj I was assuming they support both :) all right let's leave it then :)